allennlp/service/server_sanic.py

```diff
@@ -69,9 +71,13 @@ def _caching_prediction(model: Predictor, data: str) -> JsonDict:
         """
         return model.predict_json(json.loads(data))
 
-    @app.route('/predict/<model_name>', methods=['POST'])
+    @app.route('/predict/<model_name>', methods=['POST', 'OPTIONS'])
     async def predict(req: request.Request, model_name: str) -> response.HTTPResponse:  # pylint: disable=unused-variable
         """make a prediction using the specified model and return the results"""
+
+        if req.method == "OPTIONS":
```

Comment:
What is this doing?

Reply:
First off, CORS is pretty weird overall.  It doesn't prevent anything outright--it just makes it a bit harder to do.  This keeps bad javascript coders from hacking too easily.

As I understand it, when making a cross-origin request the browser first sends an OPTION request to retrieve CORS headers.  I need to handle this request specifically, because the rest of the method expects a POST body and sends the results to AllenNLP.